Add example for list-import-errors as a CI check (#32811)
mistercrunch/Airflow: docs/apache-airflow/howto/usage-cli.rst

    airflow_db={"conn_type": "mysql", "login": "root", "password": "plainpassword", "host": "mysql", "schema": "airflow"}
    druid_broker_default={"conn_type": "druid", "host": "druid-broker", "port": 8082, "extra": "{\"endpoint\": \"druid/v2/sql\"}"}

Testing for DAG Import Errors
-----------------------------
The CLI can be used to check whether any discovered DAGs have import errors via the `dags` `list-import-errors` command. The output is predictable for situations where there are no errors, such as `[]` for the json style output, so it is possible to create an automation step which fails if any DAGs cannot be imported. For instance, the check could be run in CI or pre-commit.

Example command that fails if there are any errors:
```bash
airflow dags list-import-errors | grep -q "No data found"
```

This line can be added to automation as-is, or if you want to include the output you can tee the output:

```bash
airflow dags list-import-errors | tee import_errors.txt && grep -q "No data found" import_errors.txt
```

> Exampe using jq in a Jenkins pipeline:

Reviewer: ```suggestion
Example using jq in a Jenkins pipeline:
```

    airflow_db={"conn_type": "mysql", "login": "root", "password": "plainpassword", "host": "mysql", "schema": "airflow"}
    druid_broker_default={"conn_type": "druid", "host": "druid-broker", "port": 8082, "extra": "{\"endpoint\": \"druid/v2/sql\"}"}

Testing for DAG Import Errors
-----------------------------

> The CLI can be used to check whether any discovered DAGs have import errors via the `dags` `list-import-errors` command. The output is predictable for situations where there are no errors, such as `[]` for the json style output, so it is possible to create an automation step which fails if any DAGs cannot be imported. For instance, the check could be run in CI or pre-commit.

Reviewer 2: ```suggestion
The CLI can be used to check whether any discovered DAGs have import errors via the ``list-import-errors`` subcommand. It is possible to create an automation step which fails if any DAGs cannot be imported, especially ultilizing the ``--output`` flag to produce a machine-readable format. For instance, the check could be run in CI or pre-commit.
```

Author: Incorporated a version of these changes.

Example command that fails if there are any errors:

.. code-block:: bash

>     airflow dags list-import-errors | grep -q "No data found"

Reviewer 2: I feel we should instead encourage using `--output=json` and tools such as `jq` for more robust parsing.

Author: That was actually my original approach, but i moved to this one so that i wasn't adding a new dependency. Would it make sense to add a second example using jq?

Reviewer 2: I dislike a `grep` example too since it would produce false negatives if any of the import errors contain the message *No data found*. It would be better if it checks for the string exactly, but in general working with non-structured data should not be recommended in the first place.

Author: Hmm, yeah that is a fair point.

Author: Removed the non-jq approaches.

This line can be added to automation as-is, or if you want to include the output you can tee the output:

.. code-block:: bash

    airflow dags list-import-errors | tee import_errors.txt && grep -q "No data found" import_errors.txt

Example in a Jenkins pipeline:

.. code-block:: groovy

    stage('All DAGs can load') {
        steps {
            sh 'airflow dags list-import-errors | tee import_errors.txt && grep -q "No data found" import_errors.txt'
        }
    }

> NOTE: For this to work accurately, you must ensure Airflow does not log any additional text to stdout. For example, you may need to fix any deprecation warnings, or you may need to set `lazy_load_plugins = True` in the Airflow config if you have a plugin that logs when loaded.

Reviewer 2: ```suggestion
.. note::

    For this to work accurately, you must ensure Airflow does not log any additional text to stdout. For example, you may need to fix any deprecation warnings, or need to set ``lazy_load_plugins = True`` in the Airflow config if you have a plugin that logs when loaded.
```

I think deprecation warnings log to stderr and can be avoided with `2>/dev/null` though?

Author: Quite possible, i fixed my deprecation warnings so i don't have an easy way to test, but i can include that as a recommendation.
